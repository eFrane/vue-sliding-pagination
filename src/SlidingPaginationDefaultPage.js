export default {
  name: 'SlidingPaginationDefaultPage',

  props: {
    ariaPageLabel: {
      type: String,
      required: true
    },
    isCurrent: {
      type: Boolean,
      required: true,
    },
    page: {
      type: Number,
      required: true
    },
    pageClass: {
      type: String,
      required: true
    }
  },

  computed: {
    pageClasses () {
      let res = this.pageClass

      if (this.isCurrent) {
        res += ' c-sliding-pagination__page--current'
      }

      return res
    }
  },

  methods: {
    goToPage (e) {
      e.preventDefault()
      e.stopPropagation()

      this.$emit('page-click', this.page)
    }
  },

  render (h) {
    return h('a', {
      class: this.pageClasses,
      attrs: {
        href: '#',
        'aria-label': this.ariaPageLabel
      },
      on: {
        click: this.goToPage
      }
    },
    this.page
    )
  }
}
