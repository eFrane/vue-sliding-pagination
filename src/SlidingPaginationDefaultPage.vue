<script>
export default {
  name: 'SlidingPaginationDefaultPage',

  props: {
    ariaPageLabel: {
      type: String,
      required: true
    },
    isCurrent: {
      type: Boolean,
      required: true
    },
    page: {
      type: Number,
      required: true
    },
    pageClass: {
      type: String,
      required: true
    }
  },

  computed: {
    currentClass () {
      return (this.isCurrent) ? 'c-sliding-pagination__page--current' : ''
    }
  },

  methods: {
    goToPage (e) {
      e.preventDefault()
      e.stopPropagation()

      this.$emit('page-click', this.page)
    }
  },

  render (h) {
    return h('a', {
      class: this.pageClass,
      attrs: {
        href: '#',
        'aria-label': this.ariaPageLabel
      },
      on: {
        click: this.goToPage
      }
    },
    this.page
    )
  }
}
</script>
