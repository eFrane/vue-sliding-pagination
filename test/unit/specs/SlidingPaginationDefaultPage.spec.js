import { mount } from '@vue/test-utils'
import SlidingPaginationDefaultPage from '@/SlidingPaginationDefaultPage.vue'

describe('SlidingPaginationDefaultPage.vue', () => {
  it('is named correctly', () => {
    expect(SlidingPaginationDefaultPage.name).toEqual('SlidingPaginationDefaultPage')
  })

  it('requires a page label', () => {
    expect(SlidingPaginationDefaultPage.props).toHaveProperty('ariaPageLabel.required')
    expect(SlidingPaginationDefaultPage.props.ariaPageLabel.required).toBe(true)
    expect(SlidingPaginationDefaultPage.props.ariaPageLabel.type).toBe(String)
  })

  it('requires a page class', () => {
    expect(SlidingPaginationDefaultPage.props).toHaveProperty('pageClass.required')
    expect(SlidingPaginationDefaultPage.props.ariaPageLabel.required).toBe(true)
    expect(SlidingPaginationDefaultPage.props.ariaPageLabel.type).toBe(String)
  })

  it('requires a page number', () => {
    expect(SlidingPaginationDefaultPage.props).toHaveProperty('page.required')
    expect(SlidingPaginationDefaultPage.props.page.required).toBe(true)
    expect(SlidingPaginationDefaultPage.props.page.type).toBe(Number)
  })

  it('requires a current toggle', () => {
    expect(SlidingPaginationDefaultPage.props).toHaveProperty('isCurrent.required')
    expect(SlidingPaginationDefaultPage.props.isCurrent.required).toBe(true)
    expect(SlidingPaginationDefaultPage.props.isCurrent.type).toBe(Boolean)
  })

  it('renders a link', () => {
    const wrapper = mount(SlidingPaginationDefaultPage, {
      propsData: {
        ariaPageLabel: 'Page label',
        isCurrent: false,
        page: 1,
        pageClass: 'c-sliding-pagination__page'
      }
    })

    expect(wrapper.find('a').attributes('aria-label')).toBe('Page label')
    expect(wrapper.find('a').classes('c-sliding-pagination__page')).toBeTruthy()
    expect(wrapper.text()).toBe('1')
  })

  it('changes currentClass modifier class on change of isCurrent', () => {
    const wrapper = mount(SlidingPaginationDefaultPage, {
      propsData: {
        ariaPageLabel: 'Page label',
        isCurrent: false,
        page: 1,
        pageClass: 'c-sliding-pagination__page'
      }
    })

    expect(wrapper.vm.currentClasses).toBe('c-sliding-pagination__page')
    expect(wrapper.element).toMatchSnapshot()

    wrapper.setProps({ isCurrent: true })

    expect(wrapper.vm.currentClasses).toBe('c-sliding-pagination__page c-sliding-pagination__page--current')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits a page-click on link activation', () => {
    const wrapper = mount(SlidingPaginationDefaultPage, {
      propsData: {
        ariaPageLabel: 'Page label',
        isCurrent: false,
        page: 1,
        pageClass: 'c-sliding-pagination__page'
      }
    })

    wrapper.find('a').trigger('click')
    expect(wrapper.emitted()['page-click']).toBeTruthy()
    expect(wrapper.emitted()['page-click'].length).toBe(1)
    expect(wrapper.emitted()['page-click'][0]).toEqual([1])
  })
})
