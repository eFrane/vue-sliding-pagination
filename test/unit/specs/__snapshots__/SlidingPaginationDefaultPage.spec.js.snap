// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes currentClass modifier class on change of isCurrent 1`] = `
<a
  aria-label="Page label"
  class="c-sliding-pagination__page"
  href="#"
>
  1
</a>
`;

exports[`changes currentClass modifier class on change of isCurrent 2`] = `
<a
  aria-label="Page label"
  class="c-sliding-pagination__page"
  href="#"
>
  1
</a>
`;
